docs/adr/adr_003_bundling_collection_data.md

```diff
@@ -0,0 +1,230 @@
+- Bad, because this would be a remote-settings specific job and feature, and not part of `kinto-attachment`
+
+### Option 4 - Add new API endpoint to build or serve last bundle
+With this option, we introduce a new endpoint that builds the bundle if it does not exist, or serves the one in cache if it does.
+
+If we would like to prevent multiple pods from building the same bundle, we could have a singleton “bundle-builder” container that would be dedicated to building all attachment bundles. The API would receive a request, see that the bundle is missing, and ping the bundle-builder to make it. Once the bundle-builder is done and the file is published, it would respond to all waiting API nodes so they can redirect.
+
+More alternatively, we could use event-driven architecture. The API would submit a request for the bundle to be built to a queue. A serverless function would process the request and return the response. In order to prevent multiple to build the same bundle multiple times, parallelism would have to be restricted, or we’d need to use a queue that supports partitioning based on collection name (or similar).
+- Good, because it all happens in one endpoint and easy to reason about
+- Good, because uploading/removing attachments or approving changes would not take longer
+- Bad, because a new specific container that has to be monitored, troubleshot, and deployed
+- Bad, because we have to introduce some complexity in order to prevent building several bundles in parallel
+
+### Option 5 - Build bundles on schedule
+A scheduled job iterates through the collections with the `attachment.bundle == true` flag and operates with the following logic:
+- If there is a bundle for the current collection timestamp, nothing to do.
+- If there isn’t, pull all attachments from storage, build a bundle with all current attachments, store it at `{folder}/attachments.zip` in cloud storage
+- (repeat for each bucket, including preview buckets)
+
+And creates a records bundle with all collections data (~6MB) for each bucket (`{folder}/changesets.zip`).
+
+*Note: the job that is currently in charge of publishing the highest timestamp to the Push notification server, could first execute this attachment bundling job, in order to make sure bundles will be available when clients will pull changesets.*
+
+This job would require GCloud credentials (Service Accounts credentials JSON file) and unlike `kinto-attachment` would be vendor-specific (ie. GCP Cloud storage using the `google-cloud-storage` Python library).
+
+We would have to duplicate some of the `kinto-attachment` configuration values (eg. `kinto.attachment.gcloud.bucket_name` and `kinto.attachment.folder`) in order to store the bundles along attachments (ie. in the right folder `{bucket id}/{collections id}`), so that the final bundle URL remains straightforward for clients (eg. `capabilities.attachments.base_url` + `bucket_id/collection_id /bundle.zip`)
```

Comment:
```suggestion
We would have to duplicate some of the `kinto-attachment` configuration values (eg. `kinto.attachment.gcloud.bucket_name` and `kinto.attachment.folder`) so that the final bundle URL remains straightforward for clients (eg. `capabilities.attachments.base_url` + `bundles/{bucket-id}--{collection-id}.zip`)
```